frappe/public/js/frappe/form/grid.js

```diff
@@ -212,17 +212,16 @@ export default class Grid {
 	}
 
 	refresh_remove_rows_button() {
-		this.remove_rows_button.toggleClass('hidden',
-			this.wrapper.find('.grid-body .grid-row-check:checked:first').length ? false : true);
-		this.remove_all_rows_button.toggleClass('hidden',
-			this.wrapper.find('.grid-heading-row .grid-row-check:checked:first').length ? false : true);
+		let is_visible = this.wrapper.find('.grid-body .grid-row-check:checked:first').length ? false : true;
```

Comment:
The visibility condition for `remove_rows_button` and `remove_all_rows_button` is different since the Delete All button should only be visible if the checkbox on the heading row is checked. The issue should be fixed by unchecking the checkbox on the heading row when Delete All is clicked.

Reply:
Then Delete button should not be visible when heading row should be check. Fixing that then

Reply:
@hrwX This won't work when there are multiple pages since row_count will be `false`. Why not leave the conditions as they are and uncheck the heading row checkbox when Delete or Delete All is clicked instead? Anyway it should not remain checked.